Sources/SwiftPlot/Histogram.swift

```diff
@@ -276,111 +276,79 @@ extension Histogram: HasGraphLayout {
-                    topLeft = Point(topLeft.x, currentHeight + series.scaledBinFrequency[i])
-                    topRight = Point(topRight.x, currentHeight + series.scaledBinFrequency[i])
-                    if (series.scaledBinFrequency[i] > 0) {
-                        renderer.drawLine(startPoint: topLeft,
-                                          endPoint: topRight,
-                                          strokeWidth: strokeWidth,
-                                          strokeColor: series.color,
-                                          isDashed: false)
-                        if (i != histogramSeries.bins-1) {
-                            var nextHeight = histogramSeries.scaledBinFrequency[i+1]
-                            for k in histogramStackSeries {
-                                nextHeight += k.scaledBinFrequency[i+1]
-                            }
-                            let nextTopLeft = Point(topRight.x, nextHeight)
-                            renderer.drawLine(startPoint: topRight,
-                                              endPoint: nextTopLeft,
-                                              strokeWidth: strokeWidth,
-                                              strokeColor: series.color,
-                                              isDashed: false)
-                        }
+                seriesHeights.append(currentHeights)
+            }
+            
+            // Draw only the line segments that will actually be visible, unobstructed from other lines that will be on top
+            // We iterate over the series in reverse to draw them from back to front
+            var seriesHeightsSlice = seriesHeights.reversed()[...]
+            var heightsS1Slice = seriesHeightsSlice.removeFirst()[...]
+            for (heightsSeries2, series) in zip(seriesHeightsSlice, allSeries.reversed()) {
+                var heightsS2Slice = heightsSeries2[...]
+                
+                // Grab 2 heights from the first series and 2 from the second one (iterate over bar edges of two adjacent series simultaneously)
+                var height1S1 = heightsS1Slice.removeFirst()
+                var height1S2 = heightsS2Slice.removeFirst()
+                var line = [Point]()
+                for ((height2S1, height2S2), x) in zip(zip(heightsS1Slice, heightsS2Slice), xValues) {
+                    
+                    func endLine() {
+                        renderer.drawPlotLines(points: line, strokeWidth: strokeWidth,
+                                               strokeColor: series.color, isDashed: false)
+                        line.removeAll(keepingCapacity: true)
                     }
-                    currentHeight += series.scaledBinFrequency[i]
+                    
+                    // Conditions for appending specific points or ending the line at different places based on the relative heights (4 measures)
+                    let c1 = height1S1 > height1S2
```

Comment:
I feel these might be easier to understand if `height1S1` and `height1S2` had clearer names

Reply:
I will think about these some more.